### PR TITLE
fix issue 20042 - __vector CTFE crashes the compiler

### PR DESCRIPTION
**Pull request by ghost**
This fix prevents the crash and a proper error message is issued, but doesn't add the code to accept the assign from slice to vector. This means that if this PR gets accepted then the issue would have to be reopened and converted to "enhancement".

### CODE REVIEW

File: test/fail_compilation/ice20042.d [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+DISABLED: freebsd32 linux32 osx32 win32
+TEST_OUTPUT:
+---
+fail_compilation/ice20042.d(16): Error: slice operation `cast(__vector(float[4]))nanF = [1.00000F, 2.00000F, 3.00000F, 4.00000F][0..4]` cannot be evaluated at compile time
+fail_compilation/ice20042.d(22):        called from here: `Vec4(cast(__vector(float[4]))[nanF, nanF, nanF, nanF]).this([1.00000F, 2.00000F, 3.00000F, 4.00000F])`
+---
+*/
+void write(T...)(T t){}
+
+struct Vec4 {
```
Comment (JinShil) [inline test/fail_compilation/ice20042.d]: Newline before opening curly brace according to the D Style, even in tests.

**Comment by tsbockman** [comment]
This solution is fine by me; thanks for working on this.

Please have a look at [issue 20041](https://issues.dlang.org/show_bug.cgi?id=20041), as well - it's even worse, but I suspect it has the same root cause. If this PR can fix both, a separate test case should be added for 20041, since its symptom is different.

**Comment by Basile-z** [comment]
The other issue is a 2.085 regression. This one crashed since much longer.

**Review by thewilsonator — APPROVED** [review]

**Review by ibuclaw — DISMISSED** [review]
There is no need to change formatting here.

**Pull request merged**